### PR TITLE
Halo2 common

### PR DESCRIPTION
**Pull request by tyshko-rostyslav**
Common code for 2 Halo2 benchmarks

### CODE REVIEW

File: halo2-common/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,280 @@
+// Based upon https://github.com/privacy-scaling-explorations/snark-verifier/blob/main/snark-verifier/examples/evm-verifier-with-accumulator.rs
+
+use halo2_proofs::halo2curves as halo2_curves;
+
+pub mod application {
+    use super::halo2_curves::bn256::Fr;
```
Comment (jonesmarvin8) [inline halo2-common/src/lib.rs:6]: Line 3 could be deleted. And line 6 could 'simply' be
```suggestion
    use halo2_proofs::halo2curves::bn256::Fr;
```
super reference only used on line 6, and halo2_curves labeling only used on line 6.

File: halo2-common/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+                message_arr: [Fr::ZERO; L],
+                message: Value::unknown(),
+                _spec: PhantomData,
+            }
+        }
+
+        fn configure(meta: &mut ConstraintSystem<Fr>) -> Self::Config {
+            let state = (0..WIDTH).map(|_| meta.advice_column()).collect::<Vec<_>>();
+            let _ = meta.instance_column();
+            // // meta.enable_equality(expected);
```
Comment (jonesmarvin8) [inline halo2-common/src/lib.rs:62]: Line 62, unnecessary commented out code.

File: halo2-common/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+            let partial_sbox = meta.advice_column();
+
+            let rc_a = (0..WIDTH).map(|_| meta.fixed_column()).collect::<Vec<_>>();
+            let rc_b = (0..WIDTH).map(|_| meta.fixed_column()).collect::<Vec<_>>();
+
+            meta.enable_constant(rc_b[0]);
+
+            Self::Config {
+                input: state[..RATE].try_into().unwrap(),
+                // expected,
```
Comment (jonesmarvin8) [inline halo2-common/src/lib.rs:72]: Line 72, unnecessary commented out code.

File: halo2-common/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+[package]
+name = "halo2-common"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
```
Comment (jonesmarvin8) [inline halo2-common/Cargo.toml:8]: Libraries from lines 9, 10, 11, 12, 13, and 15 do not appear to be used for lib.rs

File: halo2-common/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+itertools = "0.10.3"
+lazy_static = "1.4.0"
+num-bigint = "0.4.3"
+num-integer = "0.1.45"
+num-traits = "0.2.15"
+rand = "0.8"
+rand_chacha = "0.3.1"
+
+halo2_gadgets = { git = "https://github.com/privacy-scaling-explorations/halo2", tag = "v2023_04_20" } # not optional for now
+pasta_curves = "0.5"
```
Comment (jonesmarvin8) [inline halo2-common/Cargo.toml:18]: Line 18 dependency is not used.

File: halo2-common/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+# system_halo2
+halo2_proofs = { git = "https://github.com/privacy-scaling-explorations/halo2", tag = "v2023_04_20" } # not optional for now
+halo2curves = { git = 'https://github.com/privacy-scaling-explorations/halo2curves', tag = "0.3.2" }  # must be same version as in halo2_proofs
+
+# loader_halo2
+halo2_wrong_ecc = { git = "https://github.com/privacy-scaling-explorations/halo2wrong", tag = "v2023_04_20", package = "ecc", optional = true }
+poseidon = { git = "https://github.com/privacy-scaling-explorations/poseidon", tag = "v2023_04_20", optional = true }
+
+ark-std = { version = "0.3.0", features = ["print-trace"] }
```
Comment (jonesmarvin8) [inline halo2-common/Cargo.toml:32]: Line 32 and 33 do not appear to be used.

**Review by jonesmarvin8 — APPROVED** [review]
The comments I've provided are the only things I see in this PR that should be addressed. Mostly minor.

File: halo2-common/Cargo.toml (hunk 5 above)
Comment (tyshko-rostyslav) [reply halo2-common/Cargo.toml]: Removed

File: halo2-common/Cargo.toml (hunk 6 above)
Comment (tyshko-rostyslav) [reply halo2-common/Cargo.toml]: Removed

File: halo2-common/Cargo.toml (hunk 4 above)
Comment (tyshko-rostyslav) [reply halo2-common/Cargo.toml:8]: Removed

File: halo2-common/src/lib.rs (hunk 2 above)
Comment (tyshko-rostyslav) [reply halo2-common/src/lib.rs]: Removed

File: halo2-common/src/lib.rs (hunk 1 above)
Comment (tyshko-rostyslav) [reply halo2-common/src/lib.rs]: Removed

File: halo2-common/src/lib.rs (hunk 3 above)
Comment (tyshko-rostyslav) [reply halo2-common/src/lib.rs]: Removed

**Pull request merged**